From 01d6810d7374dfcc1e82bbbf4f7b64a255a97dfe Mon Sep 17 00:00:00 2001
From: Lukasz Tekieli <lukasz.tekieli@bmw.de>
Date: Mon, 9 Feb 2026 09:29:48 +0100
Subject: [PATCH 2/3] Add mutlicast config

---
 src/daemon/dlt.conf | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/daemon/dlt.conf b/src/daemon/dlt.conf
index 736f417..363cb0e 100644
--- a/src/daemon/dlt.conf
+++ b/src/daemon/dlt.conf
@@ -183,12 +183,12 @@ PrintASCII = 1
 # Connect with DLT Viewer UDP option to get messages by UDP socket.  
 
 # Activate UDP multicast streaming by setting this option to 1 (Default: OFF)
-# udpStreamActive = 0
+udpStreamActive = 1
 
 # Multicast/Broadcast address to which the daemon send the messages
 # Parameter is only used if udpStreamActive = 1. Only IPv4 is supported.
-# udpStreamIP = 224.0.0.1
+udpStreamIP = 224.0.0.1
 
 # UDP port to which the daemon send the messages.
 # Parameter is only used if udpStreamActive = 1.
-# udpStreamPort = 3491
+udpStreamPort = 3490
-- 
2.43.0

